LibertyDSNP/spec
# DIP-225 Social Graph / DIP-227 User Data Operations

Problem
=======
DSNP needs a way of specifying various user-related data that allows an implementation flexibility in storing social graph data while standardizing data formats and application layer data interaction semantics. GitHub issues: #225 #227 

Solution
========
The proposed user data spec strikes a compromise between the need for blockchains to segment data in optimal ways and the desire, from an application developer's point of view, to have data in an easy to work with format, whilst specifying details of schema, compression and encryption to ensure interoperability.

The user data types for the social graph that follow this approach are public follows, private follows, private connections, and private connection PRIds, and the spec describes how each is generated and interpreted.

Change summary:
---------------
- [ ] Added new page on User Data and included in navigation links
- [ ] Added items to Operations and Records tables
- [ ] Rewrote Graph page to define relevant user data types and the PRId algorithm

Steps to Verify:
----------------
1. Generate HTML with `npm run build`
2. Review spec site for content and formatting
1. Check for errors with `npm run lint`
